tgui/packages/tgui/interfaces/OutfitEditor.js

```diff
@@ -0,0 +1,138 @@
+import { Window } from '../layouts';
+
+export const OutfitEditor = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { outfit, saveable } = data;
+
+  return (
+    <Window
+      width={380}
+      height={625}>
+      <Window.Content>
+
+        <Box
+          opacity={0.5}
+          py={3}
+          position="absolute"
+          as="img"
+          src={`data:image/jpeg;base64,${data.dummy64}`}
+          height="100%"
+          width="100%"
+          style={{
+            '-ms-interpolation-mode': 'nearest-neighbor',
+          }} />
+
+        <Section
+          fill
+          title={
+            <>
+              {outfit.name}
+              <Button
+                ml={0.5}
+                color="transparent"
+                icon="pencil-alt"
+                title="Rename this outfit"
+                onClick={() => act("rename", {})} />
+            </>
+          }
+          buttons={
+            <>
+              <Button
+                icon="code"
+                tooltip="Edit this outfit on a VV window"
+                tooltipPosition="left"
+                onClick={() => act("vv", {})}
+              />
+              <Button
+                icon="save"
+                disabled={!saveable}
+                tooltip={!!saveable && "Save this outfit to the custom outfit list"}
+                tooltipPosition="left"
+                title={!saveable && "This outfit is already on the custom outfit list. Any changes made here will be immediately applied."}
+                onClick={() => act("save", {})}
+              />
+            </>
+          }>
+
+          <Box textAlign="center">
+            <Stack mb={2}>
+              <OutfitSlot name="Headgear" slot="head" />
+              <OutfitSlot name="Glasses" slot="glasses" />
+              <OutfitSlot name="Ears" slot="ears" />
+            </Stack>
+            <Stack mb={2}>
+              <OutfitSlot name="Neck" slot="neck" />
+              <OutfitSlot name="Mask" slot="mask" />
+            </Stack>
+            <Stack mb={2}>
+              <OutfitSlot name="Uniform" slot="uniform" />
+              <OutfitSlot name="Suit" slot="suit" />
+              <OutfitSlot name="Gloves" slot="gloves" />
+            </Stack>
+            <Stack mb={2}>
+              <OutfitSlot name="Suit Storage" slot="suit_store" />
+              <OutfitSlot name="Back" slot="back" />
+              <OutfitSlot name="ID" slot="id" />
+            </Stack>
+            <Stack mb={2}>
+              <OutfitSlot name="Belt" slot="belt" />
+              <OutfitSlot name="Left Hand" slot="l_hand" />
+              <OutfitSlot name="Right Hand" slot="r_hand" />
+            </Stack>
+            <Stack mb={2}>
+              <OutfitSlot name="Shoes" icon="socks" slot="shoes" />
+              <OutfitSlot name="Left Pocket" slot="l_pocket" />
+              <OutfitSlot name="Right Pocket" slot="r_pocket" />
+            </Stack>
```

Comment:
this code above here is pretty dope through and through, i hope you give each slot an icon like you did with the shoes

Reply:
Thanks - and absolutely, that's the plan.